2996_reflection/reflection.md

```diff
@@ -4615,6 +4634,49 @@ template <reflection_range R1 = span<info const>, reflection_range R2 = span<inf
 :::
 :::
 
+
+### [meta.reflection.define_class] Reflection class definition generation  {-}
+
+
+```cpp
+consteval info data_member_spec(info type,
+                                data_member_options_t options = {});
+```
+[1]{.pnum} *Mandates*: `type` designates a type.
+
+[#]{.pnum} *Returns*: A reflection of a description of the declaration of nonstatic data member with a type designated by `type` and optional characteristics designated by `options`.
+
+[#]{.pnum} *Remarks*: The reflection value being returned is only useful for consumption by `define_class`.  No other function in `std::meta` recognizes such a value.
+
+
+```c++
+  template <reflection_range R = span<info const>>
+  consteval info define_class(info class_type, R&&  mdescrs);
+```
+
+[#]{.pnum} Let `$d1$`, `$d2$`, ..., `$dN$` denote the reflection values of the range `mdescrs` obtained by calling `data_member_spec` with `type` values `$t1$`, `$t2$`, ... `$tN$` and `option` values `$o1$`, `$o2$`, ... `$oN$` respectively.  
+
+[#]{.pnum} *Mandates*: `class_type` designates an incomplete class type.  `mdescrs` is a (possibly empty) range of reflection values obtained by calls to `data_member_spec`.  `$t1$`, `$t2$`, ... `$tN$` designate types that are valid types for data members.  If `$oK$.width` (for some `$K$`) contains a value `$w$`, the corresponding type `$tK$` is a valid type for bit field of width `$w$`.  If `$oK$.alignment` (for some `$K$`) contains a value `$a$`, `alginas($a$)` is a valid `$alignment-specifier$` for a nonstatic data member of type `$tK$`.
+
+[#]{.pnum} [For example, `class_type` could be a specialization of a class template that has not been instantiated or explicitly specialized.]{.note}
+
+[#]{.pnum} *Effects*:
+Defines `class_type` with properties as follows.
+If `class_type` designates a specialization of a class template, the specialization is explicitly specialized.
+Nonstatic data members are declared in the definition of `class_type` according to `$d1$`, `$d2$`, ..., `$dN$`, in that order.
+The type of the respective members are the types denoted by the reflection values `$t1$`, `$t2$`, ... `$tN$`.
+If `$oK$.width` (for some `$K$`) contains a value, the corresponding member is declared as a bit field with that value as its width.
+If `$oK$.alignment` (for some `$K$`) contains a value `$a$`, the corresponding member is aligned as if declared with `alignas($a$)`.
+If `$oK$.name` (for some `$K$`) does not contain a value, the corresponding member is declared with an implementation-defined name.
+Otherwise, the corresponding member is declared with a name corresponding to the `string` or `u8string` value that was used to initialize `$oK$.name`.
```

Comment:
I'm not sure this is quite right (since the name could've been initialized with like `"hello"`) but good enough for now